### PR TITLE
Add option to synchronize videos on import

### LINKED ISSUE #385: Import GoPro with sync

### PR DESCRIPTION
**Pull request by philipqueen**
This PR is to add an option to the `Import Videos" wizard that will run the imported videos through the `SkellySynchronize` code that will synchronize them by cross corrolating their audio streams and clipping them so they are synchronized and have the exact same number of frames. 

We'll know it works when a user can load `freemocap` select 'import videos' and load in a folder full of (say) GoPro videos that get synchronized and saved to a `sycnhronzied_videos` folder to be processed just like any other freemocap recording. 

Tasks to be done for this to work :

## freemocap side
- [x] add a `synchronize videos by their audio track` checkbox (or parameter tree?) or a paramter tree pops up after they click "import" if that button is checked?) to the `Import Videos` wizard.

## SkellySynchronize side
- [x] save out some kind of diagnostic information to a TOML 
- [x] OPTIONAL - Make some kind of visual debug output so that we can check quality
- [ ] OPTIONAL - - Run via parameter tree model  (for now just run with default values)
- [x] OPTIONAL - fix the phone video resoultion bug (but focus on GoPro) - mostly fixed, will sometimes flip a vertical video upside down

skelly_synchronize repo: https://github.com/freemocap/skelly_synchronize
this would close #385

### CODE REVIEW

**Comment by philipqueen** [comment]
I added some additional widgets that open when you click on the synchronize box, which I think is the right option for dealing with the synchronization parameters.
Screen Shot 2023-05-04 at 11 52 32 AM

**Comment by philipqueen** [comment]
Also, I thought my last post mentioned this but if the synchronization button is clicked, it will run skelly synchronize on the videos before importing them, so this PR is minimally viable

**Comment by philipqueen** [comment]
I've changed the layout by putting all of the synchronization information in an hbox. Let me know @jonmatthis if there are any other aesthetic changes to make.  (Ignore the coloring, I'm using changes from the recoloring PR because the dialog currently doesn't show up on my computer without them)

I've also added proper error handling to skelly_synchronize if videos don't have an audio file.

Screen Shot 2023-05-09 at 4 10 41 PM

**Comment by philipqueen** [comment]
As of today this is non-blocking (runs the synchronization in a thread) and it prints the logging messages from skelly_synchronize in the console.
Screen Shot 2023-06-20 at 2 52 55 PM

File: pyproject.toml
Comment (aaroncherian) [inline pyproject.toml:1]: Mm just double checking - I saw Jon's PR a couple days ago where he unpinned the mediapipe version. Will this repin it?

**Review by aaroncherian — COMMENTED** [review]
I got this to work successfully on two long GoPro videos, as seen below:

debug_plot

**Review by jonmatthis — APPROVED** [review]
Woo!

**Pull request merged**